Forgot password and Reset password test case

llD3vi1ll opened the pull request:
Forgot password and Reset password test case are ready, and optimized.

llD3vi1ll committed:
Forgot password and Reset password test case

Forgot password and Reset password test case are ready, and optimized.

RAKESH13595 commented on src/main/java/pages/ForgotPasswordPage.java:
```diff
@@ -45,4 +59,29 @@ public Boolean verifyAttribute(ElementLocator element, String attribute, String
+			driver = new ChromeDriver();
+			break;
+		}
+		}
```
Why do we need this?

llD3vi1ll replied:
It was added to check test case regarding reset password. i have removed those.

llD3vi1ll replied:
removed

RAKESH13595 commented on src/main/java/pages/ForgotPasswordPage.java:
```diff
@@ -45,4 +59,29 @@ public Boolean verifyAttribute(ElementLocator element, String attribute, String
+	public void resetPasswordLink() {
+		// Fetch the base URL and token2 from ConfigReader
+		String baseUrl = "https://to-do-ggau.onrender.com"; // Ensure this is defined in config.properties
+		String token = ConfigReader.getProperty("token2"); // Ensure this is defined in secret.properties
```
Rename token2 to understandable variable.
Ex: resetPasswordUserToken

llD3vi1ll replied:
Noted. changed the code.

llD3vi1ll replied:
Changed.

llD3vi1ll committed:
Updated changes.

token names changed, removed int driver which was not required.

llD3vi1ll committed:
Logout_Test page

Logout test case added only one as of now. in future can check other test case for dashboard page.

RAKESH13595 commented on src/main/java/pages/ForgotPasswordPage.java:
```diff
@@ -1,9 +1,13 @@
 
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
```
remove this if it is redundant.

llD3vi1ll replied:
I have removed the redundant

RAKESH13595 commented on src/main/java/pages/ForgotPasswordPage.java:
```diff
@@ -45,4 +59,16 @@ public Boolean verifyAttribute(ElementLocator element, String attribute, String
 	}
+	public void resetPasswordLink() {
+		// Fetch the base URL and token2 from ConfigReader
+		String baseUrl = "https://to-do-ggau.onrender.com"; // Ensure this is defined in config.properties
```
this should be available in config, don't use hardcoded values.

RAKESH13595 replied:
If we are changing environments, then the url will change. Hence avoid hardcoding urls.

llD3vi1ll replied:
Made changes, now getting the link from config file.

RAKESH13595 commented on src/main/java/pages/LogoutPage.java:
```diff
@@ -0,0 +1,50 @@
+
+
+
+public class LogoutPage extends LoginPageLocatorEnum {
```
Why is there LoginPageLocatorEnum  for LogoutPage? Shouldn't this be LogoutPageLocatorEnum?

llD3vi1ll replied:
Made the changes.

RAKESH13595 commented on src/main/java/pages/LogoutPage.java:
```diff
@@ -0,0 +1,50 @@
+		actionDriver.performAction(ActionType.ENTER_TEXT, ElementLocators.PASSWORD_TEXT, password);
+	}
+
+	
```
All the below 4 methods are repeating in all 4 pages, can we move it into actionDriver? And remove from all the pages.

llD3vi1ll replied:
have moved them to actiondriver.

RAKESH13595 commented on src/test/java/tests/LogoutTest.java:
```diff
@@ -0,0 +1,40 @@
+		actionDriver.performAction(ActionType.CLICK, ElementLocators.LOGOUT);
+		Log.info("Validating Successful Sign in");
+		test.info("Validating Successful Sign in");
+		Assert.assertTrue(logoutPage.verifyMessage(ElementLocators.SIGN_IN_PAGE, "Sign in to your account"));	
```
Add a step to navigate to /dashboard and verify it is still present in Login page.

llD3vi1ll replied:
i think it will fail due to cookies not stored. i will check em and update.

llD3vi1ll replied:
as i said its not taking @RAKESH13595. when we work on auth we can write this test case.

llD3vi1ll committed:
Optimized code.

Optimized code, removed redundant codes and fixed the errors.

RAKESH13595 reviewed (approved):
LGTM
